src/js/cc_script.js

```diff
@@ -11,134 +11,127 @@
             }
-        );//end of event
+        }        
 
         /*================================Functions that control slider when hovered==============================*/
 
-        $(this).find(".rightDirection").bind(//function for right navigational button mouseover
-            "mouseover", function () {
-                settings.slider.stop();
-                var rightHover = settings.slider.css("left");
-                var rightMarginHover = parseInt(rightHover);
+        function rightMouseover(){
+            $.fn.ccCarousel.stop($.fn.ccCarousel.defaults.slider);
+            var rightHover = $.fn.ccCarousel.defaults.slider.css('left');
+            var rightMarginHover = parseInt(rightHover);
 
-                settings.slider.animate({"left": "-" + settings.addition},settings.hoverTimer);
-            }
-        );//end of event
+            $.fn.ccCarousel.defaults.slider.animate({'left': '-' + $.fn.ccCarousel.defaults.addition},$.fn.ccCarousel.defaults.hoverTimer);
+        }
 
-        $(this).find(".rightDirection").bind(//function for right navigational button mouseup
-            "mouseup", function () {
-                var rightHover = settings.slider.css("left");
-                var rightMarginHover = parseInt(rightHover);
+        function rightMouseup(){
+            var rightHover = $.fn.ccCarousel.defaults.slider.css('left');
+            var rightMarginHover = parseInt(rightHover);
 
-                settings.slider.animate({"left": "-" + settings.addition},settings.hoverTimer);
-            }
-        );//end of event
+            $.fn.ccCarousel.defaults.slider.animate({'left': '-' + $.fn.ccCarousel.defaults.addition},$.fn.ccCarousel.defaults.hoverTimer);
+        }
 
-        $(this).find(".rightDirection").bind(//function for right navigational button mouseout
-            "mouseout", function () {
-                settings.slider.stop();
-            }
-        );//end of event
+        function mouseout(){
+            $.fn.ccCarousel.stop($.fn.ccCarousel.defaults.slider);
+        }
 
         /*=========================================part for Images that contain links==================================================*/
 
-        $(this).find(".slider img").on(//function that controls links
-            "click", function () {
-                if ($(this).attr("rel") == 1) {
-                    window.open($(this).attr("alt"), "_blank");
-                }
-                else {
-                    window.location = $(this).attr("alt");
-                }
+        function imageClick(){
+            if ($(this).attr('rel') == 1) {
+                window.open($(this).attr('alt'), '_blank');
+            }
+            else {
+                window.location = $(this).attr('alt');
             }
-        );//end of event
+        }
         return $(this);
     };
 
+    $.fn.ccCarousel.stop = function(slider){
+        slider.stop();
+    }
+
+    $.fn.ccCarousel.moveToRight = function(){
```

Comment:
@Mariam13 function handlers should be declared at the beginning of IIFE function , as discussed.